python/cudf/cudf/core/column/lists.py

```diff
@@ -176,6 +176,37 @@ def __init__(self, column, parent=None):
             )
         super().__init__(column=column, parent=parent)
 
+    def extract(self, index):
+        """
+        Extracts the element at the given index from each
+        row of a Series of lists
+
+        Parameters
+        ----------
+        index : int
+            position of the desired element in the list
+
+        Returns
+        -------
+        Series
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2, 3], [3, 4, 5], [4, 5, 6]])
+        >>> s.list.extract(-1)
+        0    3
+        1    5
+        2    6
+        dtype: int64
+        """
+        if abs(index) >= self.len().min():
+            # TODO: handle case when  index == -self.len().min()
+            # e.g: index= -3 should return [1,3,4]
+            # instead of IndexError
+            raise IndexError("IndexError: list index out of range")
```

Comment:
Do we need to check this in Python? What's the libcudf behavior if an index is passed that's out of bounds?

Reply:
likely yolo: https://github.com/rapidsai/cudf/issues/5505 - libcudf `gather` for example either `NULLIFIES` or `IGNORES` (default) out-of-bounds value in the gathermap.

Reply:
We could move this logic down to the Cython layer to minimize overhead; that's what we do for gather: https://github.com/rapidsai/cudf/blob/branch-0.19/python/cudf/cudf/_lib/copying.pyx#L137-L143

Reply:
I'm fine leaving this check here. Just wanted to make sure we weren't duplicating functionality that already existed in libcudf.